p2p/src/interface/p2p_interface_impl_delegation.rs

```diff
@@ -42,4 +42,12 @@ impl<T: Deref<Target = dyn P2pInterface> + DerefMut<Target = dyn P2pInterface> +
     async fn get_connected_peers(&self) -> crate::Result<Vec<ConnectedPeer>> {
         self.deref().get_connected_peers().await
     }
+
+    async fn add_reserved_node(&mut self, addr: String) -> crate::Result<()> {
```

Comment:
What's the point of the `connect` function if we have this?

Reply:
Now, thinking about it, connect can be useful to just give a "one shot"... but then we should make this distinction clear.

Reply:
I updated comments.

Reply:
Here are the updated comments:

```
    /// Try to connect to a remote node (just once).
    /// For persistent connections `add_reserved_node` should be used.
    #[method(name = "connect")]
    async fn connect(&self, addr: String) -> rpc::Result<()>;

    /// Add the address to the reserved nodes list.
    /// The node will try to keep connections open to all reserved peers.
    #[method(name = "add_reserved_node")]
    async fn add_reserved_node(&self, addr: String) -> rpc::Result<()>;

    /// Remove the address from the reserved nodes list.
    /// Existing connection to the peer is not closed.
    #[method(name = "remove_reserved_node")]
    async fn remove_reserved_node(&self, addr: String) -> rpc::Result<()>;
```